platform/core/src/services/HangingProtocolService/HangingProtocolService.js

```diff
@@ -54,6 +65,21 @@ class HangingProtocolService {
   addProtocols(protocols) {
     protocols.forEach(protocol => {
       if (this.protocols.indexOf(protocol) === -1) {
+        protocol.id = protocol.id || protocol.name;
```

Comment:
This is just a bunch of code to allow easier configuration of the hanging protocol by defining fewer bits of it.